internal/server/http/http_test.go

```diff
@@ -0,0 +1,113 @@
+package http
+
+import (
+	"fmt"
+	"github.com/valyala/fasthttp"
+	"kube-proxless/internal/cluster/fake"
+	"kube-proxless/internal/config"
+	"kube-proxless/internal/controller"
+	"kube-proxless/internal/store/inmemory"
+	"testing"
+)
+
+func TestNewHTTPServer(t *testing.T) {
+	HTTPServer := NewHTTPServer(nil)
+
+	if HTTPServer.host != fmt.Sprintf(":%s", config.Port) {
+		t.Errorf("NewHTTPServer(nil); host == %s but must be %s",
+			HTTPServer.host, fmt.Sprintf(":%s", config.Port))
+	}
+}
+
+func TestHTTPServer_Run(t *testing.T) {
+	server := NewHTTPServer(controller.NewController(inmemory.NewInMemoryStore(), fake.NewCluster()))
+	server.client = &mockFastHTTP{}
+
+	// make sure it does not panic
+	server.Run()
```

Comment:
comment - I would suggest this test to have a timeout, run server.Run as a go thread, and check if the port is actually listening.

Reply:
The server run here is  a mock. It might feel useless but it test that this function never panic (unless fast http issue itself but I don't test fast http - should i?, it does feel like unit testing if yes)

Reply:
Gotcha, in this case, you should use mock expected calls.

```
func TestSomething(t *testing.T) {

  // create an instance of our test object
  testObj := new(MyMockedObject)

  // setup expectations
  testObj.On("DoSomething", 123).Return(true, nil)

  // call the code we are testing
  targetFuncThatDoesSomethingWithObj(testObj)

  // assert that the expectations were met
  testObj.AssertExpectations(t)


}
```

You want to make sure that fasthttp got X call - then this would be a solid unit test.